private/pkg/app/appcmd/validate.go

```diff
@@ -0,0 +1,37 @@
+// Copyright 2020-2022 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package appcmd
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ValidateRemoteNotEmpty validates that the given remote address is not an empty string
+func ValidateRemoteNotEmpty(remote string) error {
```

Comment:
This should not be in appcmd - anything in `pkg` is non-buf-specific, this is something specific to buf. This should be next to other validation functions in `buf` or `bufpkg`.

Reply:
Moved to `private/bufpkg/bufmodule/bufmoduleref/validate.go`